includes/data-stores/class-wc-product-data-store-cpt.php

```diff
@@ -174,6 +173,7 @@ public function read( &$product ) {
 		$this->read_product_data( $product );
 		$this->read_extra_data( $product );
 		$product->set_object_read( true );
+		wp_cache_set( 'product-' . $product->get_id(), $product, 'products' );
```

Comment:
What if we kept the cache set in the product factory? i.e. construct it, cache it, return it. That keeps the new caching code in one location.

Reply:
@mikejolley great suggestion :)